packages/dx-react-scheduler/src/plugins/all-day-panel.tsx

```diff
@@ -36,9 +53,11 @@ export class AllDayPanel extends React.PureComponent {
     };
     this.allDayPanelRef = this.allDayPanelRef.bind(this);
 
-    this.appointmentPlaceholder = params => <TemplatePlaceholder name="appointment" params={params} />;
+    this.appointmentPlaceholder = params =>
```

Comment:
These placeholders do not use instance members so should be extracted out of the class

Reply:
We should keep these methods here because when we use a few similar plugins we have problems with re-rendering.

It's as relevant for view plugins. And we did it same in the AllDayPanel plugin.

Reply:
> problems with re-rendering
That's quite odd. May be that render issue should be fixed itself?

Reply:
What do you mean @DmitryBogomolov?

Reply:
Odd is the fact that this component

```jsx
params => <TemplatePlaceholder name="appointment" params={params} />
```

does not depend on instance but causes render issues when moved out of instance.